oxidation/libparsec/crates/types/src/time.rs

```diff
@@ -45,12 +45,11 @@ impl DateTime {
         minute: u32,
         second: u32,
         microsecond: u32,
-    ) -> Self {
-        Self(
-            chrono::Utc
-                .ymd(year, month, day)
-                .and_hms_micro(hour, minute, second, microsecond),
-        )
+    ) -> LocalResult<Self> {
+        chrono::Utc
+            .with_ymd_and_hms(year, month, day, hour, minute, second)
+            .map(|x| x + Duration::microseconds(microsecond as i64))
```

Comment:
Shouldn't we take `microsecond` as a 64bits integer here? It will avoid integer overflow

Reply:
Oh, mb I looked back and this cast is totally safe